sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -295,7 +287,6 @@ private[sql] class ParquetRelation2(
           val inputFormat = if (cacheMetadata) {
             new FilteringParquetRowInputFormat {
               override def listStatus(jobContext: JobContext): JList[FileStatus] = cachedStatuses
-              override def getFooters(jobContext: JobContext): JList[Footer] = cachedFooters
```

Comment:
Removed in order to prevent eager evaluation of footers. However, I'm not even sure this code is helpful. Could some provide a real world scenario where it would be useful? No performance hit in our workload.